docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc

```diff
@@ -107,104 +378,224 @@ public class ProtectedResource {
+=== Handling and controlling the life time of authentication
+
+Another important aspect of authentication is to be sure that data it is based on is fresh while not requiring the user to authenticate at every single request.
+There are also situation when a logout is explicitly requested.
+This section discusses the various key notions necessarity to find the right balance for your Quarkus application.
 
 [[oidc-cookies]]
-=== Cookies
+==== Cookies
 
 The OIDC adapter uses cookies to keep the session, code flow, and post-logout state.
+This state is a key element controlling the life time of authentication data.
 
-`quarkus.oidc.authentication.cookie-path` property is used to ensure the cookies are visible especially when you access the protected resources with overlapping or different roots, for example:
+`quarkus.oidc.authentication.cookie-path` property is used to ensure the same cookie is visible when you access the protected resources with overlapping or different roots, for example:
 
 * `/index.html` and `/web-app/service`
 * `/web-app/service1` and `/web-app/service2`
 * `/web-app1/service` and `/web-app2/service`
 
-`quarkus.oidc.authentication.cookie-path` is set to `/` by default but can be narrowed to the more specific root path such as `/web-app`.
+`quarkus.oidc.authentication.cookie-path` is set to `/` by default but can be narrowed to a more specific root path such as `/web-app`.
 
-You can also set a `quarkus.oidc.authentication.cookie-path-header` property if the cookie path needs to be set dynamically.
+Sometimes the cookie path needs to be set dynamically.
+Set the `quarkus.oidc.authentication.cookie-path-header` property.
 For example, setting `quarkus.oidc.authentication.cookie-path-header=X-Forwarded-Prefix` means that the value of HTTP `X-Forwarded-Prefix` header will be used to set a cookie path.
 
 If `quarkus.oidc.authentication.cookie-path-header` is set but no configured HTTP header is available in the current request then the `quarkus.oidc.authentication.cookie-path` will be checked.
 
-If your application is deployed across multiple domains, make sure to set a `quarkus.oidc.authentication.cookie-domain` property for the session cookie be visible to all protected Quarkus services, for example, if you have 2 services deployed at:
+If your application is deployed across multiple domains, set a `quarkus.oidc.authentication.cookie-domain` property for the session cookie be visible to all protected Quarkus services, for example, if you have 2 services deployed at:
 
 * https://whatever.wherever.company.net/
 * https://another.address.company.net/
 
 then the `quarkus.oidc.authentication.cookie-domain` property must be set to `company.net`.
```

Comment:
Make active:
```suggestion
Set the `quarkus.oidc.authentication.cookie-domain` property to `company.net`.
```

Reply:
@michelle-purcell Can we have `then set` instead of `Set` ? Otherwise it is a new sentence which kind of loses the connection to the example

Reply:
@sberyozkin - I see what you mean. I updated my comment with an alternative way that includes "..then set..." in the sentence.

I also added the `\` to the domain examples to escape them, otherwise, they get flagged as broken links on the Red Hat portal.